mlflow/utils/__init__.py

```diff
@@ -193,3 +193,12 @@ def is_iterator(obj):
     :return: boolean representing whether or not 'obj' is an iterator.
     """
     return (hasattr(obj, "__next__") or hasattr(obj, "next")) and hasattr(obj, "__iter__")
+
+
+def is_in_ipython_notebook():
```

Comment:
```suggestion
def _is_in_ipython_notebook():
```

Reply:
We're pretty inconsistent about this throughout MLflow. For example, similar utils defined in `databricks_utils` don't have leading underscores, and other utility methods in this module don't have underscores.

From https://peps.python.org/pep-0008/#public-and-internal-interfaces, it appears that, to really comply with conventions, all of our internal module names (e.g. `mlflow.utils`) should be prefixed with `_` too (e.g. `mlflow._utils`). We can revisit this later.

Reply:
If we were to make such a change, we'd probably want to avoid doing it until MLflow 2.0. For better or worse, a lot of other libraries depend on our internal functions.

Reply:
Agree, I personally don't like this prefix-underscore-to-indicate-private convention. It's challenging to be consistent.